dias/tracker.py

```diff
@@ -0,0 +1,400 @@
+    Path to location of files. (default: location of current staging directory.
+conf : dict
+    dictionary of configuration key:pair values for file_index.db location
+
+Example
+-------
+client = dias.tracker.Tracker('/mnt/gong/staging/', '/mnt/gong/staging/file_index.db')
+client.add_analyzer_ine('my_analyzer')
+my_todo = client.new_files('my_analyzer', filetypes="chimecal")
+self.do_things(my_todo)
+client.register_done('my_analyzer', my_todo)
+"""
+
+import re
+import os
+import glob
+import h5py
+import yaml
+import logging
+
+from collections import namedtuple
+
+from ch_util import ephemeris
+from dias import DiasUsageError
+from caput import config
+from peewee import (
+    Model,
+    CharField,
+    SqliteDatabase,
+    ForeignKeyField,
+    DateTimeField,
+    BooleanField,
+    OperationalError,
+    JOIN,
+    chunked,
+)
+
+
+# Initialize db in Tracker.__init__()
+db = SqliteDatabase(None)
+
+FILETYPES = ["chimestack", "chimecal", "chimetiming"]
+
+
+class BaseModel(Model):
+    """Base Model defining the database connection."""
+
+    class Meta:
+        """The definition of the database connection."""
+
+        database = db
+
+
+class FileType(BaseModel):
+    """
+    Model definition for table of filetypes that DIAS uses.
+
+    Attributes
+    ----------
+    name : String
+        name of filetype; matches how they would be written in directory names
+    id : Integer
+        unique id for filetype in database
+    """
+
+    name = CharField(unique=True)
+
+
+class Analyzer(BaseModel):
+    """
+    Model definition for the table of analyzers.
+
+    Attributes
+    ----------
+    name : String
+        name of analyzer
+    """
+
+    name = CharField(unique=True)
+
+
+class File(BaseModel):
+    """
+    Model definition for table of files that Dias analyzers interact with.
+
+    Attributes
+    ----------
+    filepath : String
+        Absolute path for file
+    file_type_id : Integer
+        reference to filetype of that file
+    start_time : datetime.datetime
+        timestamp of earliest datapoint in file
+    end_time : datetime.datetime
+        timestamp of latest datapoint in file
+    exists : Boolean
+        does the file still exist at filepath
+    """
+
+    filepath = CharField(unique=True)
+    file_type_id = ForeignKeyField(FileType, backref="files")
+    start_time = DateTimeField()
+    end_time = DateTimeField()
+    exists = BooleanField()
+
+
+class Processed(BaseModel):
+    """
+    Model for table of analyzer/file pairs which tracks which files have already been processed by the analyzers.
+
+    Attributes
+    ----------
+    file_id : Integer
+        Refers to a file in a file table
+    analyzer_id : Integer
+        Refers to an analyzer in the analyzer table
+    """
+
+    file_id = ForeignKeyField(File, backref="processed")
+    analyzer_id = ForeignKeyField(Analyzer, backref="processed")
+
+    class Meta:
+        """Meta for the Model."""
+
+        database = db
+
```

Comment:
```suggestion
```

Reply:
Are you sure? I had to override the `Meta` to add those indexes. Would it still know `database = db`?